file_handler.go

```diff
@@ -0,0 +1,67 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+
+	"github.com/nlopes/slack"
+)
+
+// FileHandler downloads files from slack
+type FileHandler struct {
+	SlackAPIKey          string
+	FileDownloadLocation string
+	ProxyPrefix          string
+}
+
+// Download downloads url contents to a local file
+func (handler *FileHandler) Download(file slack.File) string {
+	fileUrl := file.URLPrivate
+	if handler.FileDownloadLocation == "" {
+		return fileUrl
+	}
+	if handler.SlackAPIKey == "" {
+		return fileUrl
+	}
+	if file.IsExternal {
```

Comment:
the above `if`s could be collapsed into a longer one (Go doesn't mandate the 80 columns limitation), and this also makes the intent immediate